### DISCUSSION TITLE
Dynamic FetchXML support

### DISCUSSION THREAD

**Discussion by marcuscrm** [khoait/DCE.PCF#44] [Ideas] [answered]
First of all, this is a brilliant PCf.  The dynamic templating is very powerful.  However, in my use case, I have multiple dependent lists on one form.  It would be useful to be able to specify the FetchXML directly, including the templating placeholders, rather than create "empty" System Views that need to be bound on multiple fields on the form.

**Comment by khoait** [comment]
Thanks for your idea! I agree that this will be useful. I'll need to think about how to store the fetchxml first and let you know.

**Reply by crmvet** [reply]
Could it be a name of an Environment Variable for that filter perhaps? i.e. in PCF settings on form for each instance of PCF we spell the unique name of an environment variable that contains XML?

**Reply by khoait** [reply]
yeah. that's a good idea.

below are a few possible options to get the fetchxml:
- view name (currently)
- environment variable name
- custom configuration record. This can be in a form of an odata url to retrieve a single column. for example, `/new_myconfigurations(00000000-0000-0000-0000-000000000001)/new_value` . see more here: https://learn.microsoft.com/en-us/power-apps/developer/data-platform/webapi/retrieve-entity-using-web-api#retrieve-a-single-property-value
- bound to a field on the form. This can help with constructing a fetchxml with your own logic

the tricky part is how we can make it as simple as possible to pass in this config to the control without adding too many props. Below is one possible flow just using the current lookup view prop:

```mermaid
flowchart TD
    P[LookupView value] --> F{contains < fetch>?}
    F -->|Yes| A[use value as is]
    A --> X
    F -->|No| O{contains odata url?}
    O -->|Yes| B[use value returned from odata]
    B --> X
    O -->|No| SV[search view name]
    SV --> V{found?}
    V -->|Yes| C[use view fetchxml]
    C --> X
    V -->|No| SE[search environment variable]
    SE --> E{found?}
    E -->|Yes| D[use env fetchxml]
    D --> X
    E -->|No| X(((End)))
```
Let me know what you think.

Cheers!

**Reply by crmvet** [reply]
Yeah, makes all the sense, especially with custom record. As an environment variable host 2000 chars only and text area in a record 1048576. So for larger XMLs a custom record is way better.

**Reply by khoait** [reply]
@marcuscrm @crmvet, I created #121 to implement this.

**Reply by khoait** [reply, marked as answer]
@marcuscrm @crmvet #121 is done and will be included in the next release 1.0.0.7. cheers!
